networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/BlockGossipManager.java

```diff
@@ -13,17 +13,29 @@
 
 package tech.pegasys.teku.networking.eth2.gossip;
 
+import com.google.common.annotations.VisibleForTesting;
+import java.util.Optional;
 import tech.pegasys.teku.infrastructure.async.AsyncRunner;
 import tech.pegasys.teku.networking.eth2.gossip.encoding.GossipEncoding;
 import tech.pegasys.teku.networking.eth2.gossip.topics.GossipTopicName;
-import tech.pegasys.teku.networking.eth2.gossip.topics.OperationProcessor;
+import tech.pegasys.teku.networking.eth2.gossip.topics.OperationMilestoneValidator;
+import tech.pegasys.teku.networking.eth2.gossip.topics.TimedOperationProcessor;
+import tech.pegasys.teku.networking.eth2.gossip.topics.topichandlers.Eth2TopicHandler;
+import tech.pegasys.teku.networking.eth2.gossip.topics.topichandlers.TimedEth2TopicHandler;
 import tech.pegasys.teku.networking.p2p.gossip.GossipNetwork;
+import tech.pegasys.teku.networking.p2p.gossip.TopicChannel;
 import tech.pegasys.teku.spec.Spec;
 import tech.pegasys.teku.spec.datastructures.blocks.SignedBeaconBlock;
 import tech.pegasys.teku.spec.datastructures.state.ForkInfo;
 import tech.pegasys.teku.storage.client.RecentChainData;
 
-public class BlockGossipManager extends AbstractGossipManager<SignedBeaconBlock> {
+public class BlockGossipManager implements GossipManager {
```

Comment:
what if we change the `OperationProcessor:process` to accept a Metadata or Context class in which we store stuff. And we make it optional with a default empty implementation to be backward compatible. 
This whay we can avoid Timed* duplications and `arrivalTimestamp` will be in it. We can potentially store the peerid or something else in the future.

Or, for simplicity, we could have directly the optional timestamp for now there, that can become a context when\if we will add more stuff.

Reply:
I wanted to keep OperationProcessor and Eth2TopicHandler as is, so we don't need to change 100 of classes. I will try your suggestion to compare 2 approaches